fuel-indexer/indexer/src/api.rs

```diff
@@ -0,0 +1,127 @@
+use crate::{IndexerConfig, SchemaManager};
+use async_std::sync::{Arc, RwLock};
+use fuel_indexer_schema::db::{
+    graphql::{GraphqlError, GraphqlQueryBuilder},
+    tables::Schema,
+};
+use serde::Deserialize;
+use thiserror::Error;
+use tokio_postgres::{connect, types::Type, NoTls};
+use tracing::error;
+use warp::{http::StatusCode, reply::with_status, Filter, Reply};
+
+const MAX_JSON_BODY: u64 = 1024 * 16;
+
+#[derive(Debug, Error)]
+enum APIError {
+    #[error("Postgres error {0:?}")]
+    PostgresError(#[from] tokio_postgres::Error),
+    #[error("Query builder error {0:?}")]
+    GraphqlError(#[from] GraphqlError),
+    #[error("Unexpected DB type {0:?}")]
+    UnexpectedDBType(String),
+    #[error("Unexpected JSON type {0:?}")]
+    UnexpectedJSONType(String),
+    #[error("Serde Error {0:?}")]
+    SerdeError(#[from] serde_json::Error),
+}
+
+#[derive(Clone, Deserialize)]
+struct Query {
+    query: String,
+    #[allow(unused)] // TODO
+    params: String,
+}
+
+fn extract_json() -> impl Filter<Extract = (Query,), Error = warp::Rejection> + Clone {
+    warp::body::content_length_limit(MAX_JSON_BODY).and(warp::body::json())
+}
+
+async fn query_graph(
+    name: String,
+    query: Query,
+    manager: Arc<RwLock<SchemaManager>>,
+    config: Arc<IndexerConfig>,
+) -> Result<impl warp::Reply, warp::Rejection> {
+    match manager.read().await.load_schema(&name) {
+        Ok(schema) => match run_query(query, schema, config).await {
+            Ok(response) => Ok(with_status(response, StatusCode::OK).into_response()),
+            Err(e) => {
+                error!("Query error {e:?}");
+                Ok(with_status(
+                    "Internal Server Error".to_string(),
+                    StatusCode::INTERNAL_SERVER_ERROR,
+                )
+                .into_response())
+            }
+        },
+        Err(e) => Ok(with_status(
+            format!("The graph {} was not found ({:?})", name, e),
+            StatusCode::NOT_FOUND,
+        )
+        .into_response()),
+    }
+}
+
+pub struct GraphQlAPI;
```

Comment:
Nit: rust naming conventions says we shouldn't uppercase all letters in an acronym. 
https://rust-lang.github.io/api-guidelines/naming.html#casing-conforms-to-rfc-430-c-case

```suggestion
pub struct GraphQlApi;
```